bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/MeteoAlertIconProvider.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meteoalerte.internal;
+
+import static org.openhab.binding.meteoalerte.internal.MeteoAlerteBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteoalerte.internal.json.ResponseFieldDTO.AlertLevel;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.types.State;
+import org.openhab.core.ui.icon.IconProvider;
+import org.openhab.core.ui.icon.IconSet;
+import org.openhab.core.ui.icon.IconSet.Format;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MeteoAlertIconProvider} is the class providing binding related icons.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(immediate = true, service = { IconProvider.class })
+@NonNullByDefault
+public class MeteoAlertIconProvider implements IconProvider {
+    private static final Set<String> ICONS = Set.of(WAVE.replace("-", "_"), AVALANCHE, HEAT, FREEZE.replace("-", "_"),
+            FLOOD, SNOW, STORM, RAIN.replace("-", "_"), WIND);
+    private static final IconSet ICON_SET = new IconSet(BINDING_ID, "Météo Alerte Icons",
+            "These are the icons describing weather events", Set.of(Format.SVG));
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlertIconProvider.class);
+    private final ClassLoader classLoader;
+
+    @Activate
+    public MeteoAlertIconProvider() {
+        classLoader = Objects.requireNonNull(MeteoAlertIconProvider.class.getClassLoader());
+    }
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return Set.of(ICON_SET);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        return getIconSets();
```

Comment:
You could consider translations like in `AbstractResourceIconProvider `.

Reply:
I could but I do not get the point of translations regarding an IconProvider

Reply:
You're right if this is not displayed somewhere in Main UI.
And I think it isn't.

Reply:
Ok, got it. I'll be able to translate the IconSet description.

Reply:
> You're right if this is not displayed somewhere in Main UI. And I think it isn't.

It is only displayed in REST answers.

Reply:
Added